homeassistant/components/recorder/__init__.py

```diff
@@ -101,9 +104,13 @@ def run_information(point_in_time: Optional[datetime]=None):
             start=_INSTANCE.recording_start,
             closed_incorrect=False)
 
-    return query('RecorderRuns').filter(
-        (recorder_runs.start < point_in_time) &
-        (recorder_runs.end > point_in_time)).first()
+    try:
+        return query('RecorderRuns').filter(
+            (recorder_runs.start < point_in_time) &
+            (recorder_runs.end > point_in_time)).first()
+    finally:
+        Session().commit()
```

Comment:
I was trying to get rid of unnecessary ROLLBACKS. Try `py.test -s -k recorder 2>&1 | grep ROLLBACK`

The correct way seems to be to initialize the pool with None